plugins/ptp_operator/metrics/ptp4lParse.go

```diff
@@ -64,15 +64,15 @@ func (p *PTPEventManager) ParsePTP4l(processName, configName, profileName, outpu
 		lastRole := ptpInterface.Role
 		ptpIFace := ptpInterface.Name
 
-		if role == types.SLAVE {
+		if role == types.SLAVE || masterOffsetSource == ts2phcProcessName {
 			// initialize
 			if _, found := ptpStats[ClockRealTime]; !found {
 				ptpStats[ClockRealTime] = stats.NewStats(configName)
 				ptpStats[ClockRealTime].SetProcessName(phc2sysProcessName)
 			}
 			if _, found := ptpStats[master]; !found {
 				ptpStats[master] = stats.NewStats(configName)
-				ptpStats[master].SetProcessName(ptp4lProcessName)
+				ptpStats[master].SetProcessName(masterOffsetSource)
```

Comment:
will the masterOffsetSource be set to the correct value if we reach here for the `ptp4l`case?

Reply:
At this point, we assume that  the ptpconfig has been configured properly. In that case, we can either have ptp4l offset or ts2phcs offset reporting. we can't have both. So when offset is read for the first time from master this will be updated